hclwrite unit tests seem to be intermittently failing

Our automated test runs in Circle-CI seem to be occasionally raising a test failure in the `hclwrite` package. This failure does not seem to be consistent, so it seems likely that there's some non-determinism in the `hclwrite` logic.

```
--- FAIL: TestBodyRemoveBlock (0.00s)
    ast_body_test.go:1249: Removing the first block
    ast_body_test.go:1250: initial content:
        a = 1
        
        # Foo
        foo {
          b = 1
        }
        foo {
          b = 2
        }
        bar {}
    ast_body_test.go:1260: updated content:
        a = 1
        
        # Foo
        foo {
          b = 1
        }
        bar {}
    ast_body_test.go:1357: wrong result
        got:  (hclwrite.Tokens) (len=19 cap=32) {
         (*hclwrite.Token)(0xc000256000)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  61                                                |a|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256028)({
          Type: (hclsyntax.TokenType) TokenEqual,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  3d                                                |=|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000256050)({
          Type: (hclsyntax.TokenType) TokenNumberLit,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  31                                                |1|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000256078)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002560a0)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002560c8)({
          Type: (hclsyntax.TokenType) TokenComment,
          Bytes: ([]uint8) (len=6 cap=6) {
           00000000  23 20 46 6f 6f 0a                                 |# Foo.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002560f0)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=3 cap=3) {
           00000000  66 6f 6f                                          |foo|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256118)({
          Type: (hclsyntax.TokenType) TokenOBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7b                                                |{|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000256140)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256168)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  62                                                |b|
          },
          SpacesBefore: (int) 2
         }),
         (*hclwrite.Token)(0xc000256190)({
          Type: (hclsyntax.TokenType) TokenEqual,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  3d                                                |=|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc0002561b8)({
          Type: (hclsyntax.TokenType) TokenNumberLit,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  31                                                |1|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc0002561e0)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256208)({
          Type: (hclsyntax.TokenType) TokenCBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7d                                                |}|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256230)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002563c0)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=3 cap=3) {
           00000000  62 61 72                                          |bar|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002563e8)({
          Type: (hclsyntax.TokenType) TokenOBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7b                                                |{|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000256410)({
          Type: (hclsyntax.TokenType) TokenCBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7d                                                |}|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000256438)({
          Type: (hclsyntax.TokenType) TokenEOF,
          Bytes: ([]uint8) {
          },
          SpacesBefore: (int) 0
         })
        }
        
        want: (hclwrite.Tokens) (len=18 cap=18) {
         (*hclwrite.Token)(0xc000250f90)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  61                                                |a|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000250fc0)({
          Type: (hclsyntax.TokenType) TokenEqual,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  3d                                                |=|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000250ff0)({
          Type: (hclsyntax.TokenType) TokenNumberLit,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  31                                                |1|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000251020)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251050)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251080)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=3 cap=3) {
           00000000  66 6f 6f                                          |foo|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002510b0)({
          Type: (hclsyntax.TokenType) TokenOBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7b                                                |{|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc0002510e0)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251110)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  62                                                |b|
          },
          SpacesBefore: (int) 2
         }),
         (*hclwrite.Token)(0xc000251140)({
          Type: (hclsyntax.TokenType) TokenEqual,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  3d                                                |=|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000251170)({
          Type: (hclsyntax.TokenType) TokenNumberLit,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  32                                                |2|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc0002511a0)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002511d0)({
          Type: (hclsyntax.TokenType) TokenCBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7d                                                |}|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251200)({
          Type: (hclsyntax.TokenType) TokenNewline,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  0a                                                |.|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251230)({
          Type: (hclsyntax.TokenType) TokenIdent,
          Bytes: ([]uint8) (len=3 cap=3) {
           00000000  62 61 72                                          |bar|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc000251260)({
          Type: (hclsyntax.TokenType) TokenOBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7b                                                |{|
          },
          SpacesBefore: (int) 1
         }),
         (*hclwrite.Token)(0xc000251290)({
          Type: (hclsyntax.TokenType) TokenCBrace,
          Bytes: ([]uint8) (len=1 cap=1) {
           00000000  7d                                                |}|
          },
          SpacesBefore: (int) 0
         }),
         (*hclwrite.Token)(0xc0002512c0)({
          Type: (hclsyntax.TokenType) TokenEOF,
          Bytes: ([]uint8) {
          },
          SpacesBefore: (int) 0
         })
        }
        
        diff:
        {hclwrite.Tokens}[?->5]:
        	-: <non-existent>
        	+: &hclwrite.Token{Type: s"TokenComment", Bytes: []uint8{0x23, 0x20, 0x46, 0x6f, 0x6f, 0x0a}}
        {hclwrite.Tokens}[10->11].Bytes:
        	-: []uint8{0x32}
        	+: []uint8{0x31}
    ast_body_test.go:1360: removing the second block
    ast_body_test.go:1361: initial content:
        a = 1
        
        # Foo
        foo {
          b = 1
        }
        bar {}
    ast_body_test.go:1370: updated content:
        a = 1
        
        bar {}
FAIL
FAIL	github.com/hashicorp/hcl/v2/hclwrite	0.140s
```

--- a/hclwrite/ast_body.go
+++ b/hclwrite/ast_body.go
@@ -60,7 +60,7 @@ func (b *Body) Attributes() map[string]*Attribute {
 // Blocks returns a new slice of all the blocks in the body.
 func (b *Body) Blocks() []*Block {
 	ret := make([]*Block, 0, len(b.items))
-	for n := range b.items {
+	for _, n := range b.items.List() {
 		if block, isBlock := n.content.(*Block); isBlock {
 			ret = append(ret, block)
 		}

